rotary_utils/repair.py

```diff
@@ -622,105 +494,147 @@ def run_end_repair(long_read_filepath: str, assembly_fasta_filepath: str, assemb
     :param threads_mem_mb: memory in MB per thread (to use for samtools); must be an integer
     """
 
-    # Define core file names and directory structures
-    # These will be the files and folders in the main output directory:
-    end_repaired_contigs_filepath = os.path.join(output_dir, 'repaired.fasta')
-    end_repair_status_filepath = os.path.join(output_dir, 'repaired_info.tsv')
-    verbose_logfile = os.path.join(output_dir, 'verbose.log')
-    bam_filepath = os.path.join(output_dir, 'long_read.bam')
-    circlator_logdir = os.path.join(output_dir, 'circlator_logs')
-    linking_outdir_base = os.path.join(output_dir, 'contigs')
-    circular_contig_tmp_fasta = os.path.join(linking_outdir_base, 'circular_input.fasta')
+    repair_paths = RepairPaths(output_dir)
+    assembly_info = AssemblyInfo(assembly_fasta_filepath, assembly_info_filepath, assembly_info_type)
```

Comment:
Here, I instantiate the RepairPaths and AssemblyInfo objects representing various pieces of data the parent function needs. Note that the signature of `run_end_repair` contains all the data required to instantiate AssemblyInfo. This likely means that AssemblyInfo could be instantiated higher up the call chain by passing a parameter to `run_end_repair` containing a previously instantiated AssemblyInfo object.

Reply:
I moved the `AssemblyInfo()` instantiation to main and then pass an AssemblyInfo object to `run_end_repair` now. Makes the `run_end_repair` settings much more compact!

Reply:
>  I moved the docstrings for these params to __init__ inside class AssemblyInfo. Is this appropriate?

Yes